### PR TITLE
adding opls

### PR DESCRIPTION
**Pull request by jwist**
adding an alternative opls loop that match closely the results from MetaboMate R-package. It only support regression, while discriminant analysis will be added later.

### CODE REVIEW

**Comment by lpatiny** [comment]
I would homogenise upper / lower case in names.
OPLSNipal, OPLS, PLS

I would add a folder 'util' to put utils.js, tss, summaryMetadata, getFolds and their corresponding tests

In __tests__ there is no reason to have one more level for kopls and pls

**Comment by lpatiny** [comment]
Please could you update the README ?
If I unsertand well there should now be 3 methods exposed, OPLS, PLS and KOPLS

**Comment by lpatiny** [comment]
How KOPLS differs from OPLS ? Speed ? Result ?

File: package.json [hunk 1]
```diff
@@ -38,18 +39,28 @@
   },
   "homepage": "https://github.com/mljs/pls",
   "dependencies": {
-    "ml-matrix": "^6.1.2"
+    "ml": "^5.0.0",
```
Comment (targos) [inline package.json]:
```suggestion
    "ml": "^5.0.0",
```

Please remove this dependency. We should only depend on single packages like the ones below

File: src/getFolds.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// from CV.kfold ml-cross-validation
```
Comment (targos) [inline src/getFolds.js]: can you remove it and use the ml-cross-validation package now?

File: src/index.js [hunk 3]
```diff
@@ -1,2 +1,5 @@
 export { PLS } from './pls';
 export { KOPLS } from './kopls';
+export { OPLS } from './OPLS';
+export { oplsNIPALS } from './oplsNIPALS';
+export { summaryMetadata } from './summaryMetadata';
```
Comment (targos) [inline src/index.js]: I'm not sure this function makes sense in this package. It doesn't seem specific to PLS and isn't necessary to execute the algorithms

File: src/getFolds.js (hunk 2 above)
Comment (jwist) [reply src/getFolds.js]: done

File: package.json (hunk 1 above)
Comment (jwist) [reply package.json]: done

Comment (jwist) [reply package.json]: I removed dist. Should I also remove lib?

Comment (targos) [reply package.json]: > I removed dist.

I think you forgot to push

> Should I also remove lib?

Yes, but I don't see it here.

**Pull request merged**